flower_shop/models/stock_lot.py

```diff
@@ -0,0 +1,34 @@
+from odoo import fields, models, api
+from datetime import datetime, timedelta
+from odoo.exceptions import ValidationError
+
+
+class StockLot(models.Model):
+    _inherit = 'stock.lot'
+
+    flower_water_id = fields.One2many('flower.water', 'lot_id')
+
+    def water_flower(self):
+        self.write({'flower_water_id': [(0, 0, {'lot_id': self.id, 'watering_datetime': datetime.now()})]})
+
+
+    @api.constrains('flower_water_id', 'flower_water_id.watering_datetime')
+    def prevent_watering_before_time(self):
+        for record in self:
+            if record.product_id.flower_id:
+                if len(record.flower_water_id) > 1:

```

Comment:
Can be shortened into one If check: **if record.product_id.flower_id and len(record.flower_water_id) > 1**